### DISCUSSION TITLE
Ability to loop over all items in Checkboxes, Multi Select, Radio Buttons, Select Dropdown, and Selectable Buttons fieldtypes

### DISCUSSION THREAD

**Discussion by PaulBaileyAtChan** [ExpressionEngine/ExpressionEngine#2870] [Feature Requests]
EE has a number of fieldtypes which consist of a list of value:label pairs: Checkboxes, Multi Select, Radio Buttons, Select Dropdown, Selectable Buttons. Using template code, within the `{exp:channel:entries}` tag, it's possible to loop over the _chosen_ items for each of these fieldtypes for any channel entry. However, it's _not_ currently possible, using template code, to loop over all of the items in the field itself, outside of the `{exp:channel:entries}` tag.

There are lots of situations in which looping over all of the available items in a field using one of these fieldtypes, can be useful in template code outside of a channel entries tag: to generate menu items; navigation controls, content filters, etc. This type of general facility exists for categories, using the `{exp:channel:categories}` tag, but there's no analogous facility for the various value:label fieldtypes. This feels like an obvious gap.

**Comment by robinsowell** [comment]
This does indeed come up in support.  Not a ton, but frequently enough that it's a thing.

**Comment by intoeetive** [comment]
I think we would need new tag for this (ex. `{exp:channel:field}`) for this that would accept field short name or ID as parameter and display field options (and possibly other field settings? not sure)

**Comment by ic360-com** [comment]
I've come across the perfect use case for this today.
I want to output the options from a Select Dropdown field... in a form... without using Channel Forms.
Tried an SQL Query, but it looks like the Options are encoded in the field_settings column of the exp_channel_fields table.

**Comment by intoeetive** [comment]
https://github.com/ExpressionEngine/ExpressionEngine/pull/3809
